packages/vscode-extension/src/commands/CreateFile.ts

```diff
@@ -0,0 +1,209 @@
+
+interface SuiteScriptTypeItem extends QuickPickItem {
+	id: string;
+	name: string;
+}
+
+const COMMAND_NAME = 'createfile';
+
+export default class CreateFile extends BaseAction {
+	constructor() {
+		super(COMMAND_NAME);
+	}
+
+	protected validate(): { valid: false; message: string } | { valid: true } {
+		const superValidation = super.validate();
+		if (!superValidation.valid) {
+			return superValidation;
+		}
+
+		const folderChoices = this.getFolderChoices();
+		if (folderChoices.length === 0) {
+			return {
+				valid: false,
+				message: this.translationService.getMessage(
+					CREATE_FILE.ERRORS.MISSING_VALID_FOLDER_FOR_SUITECRIPT_FILE,
+					this.vscodeCommandName,
+					ApplicationConstants.LINKS.INFO.PROJECT_STRUCTURE
+				),
+			};
+		}
+
+		return {
+			valid: true,
+		};
+	}
+
+	protected async execute(): Promise<void> {
+		const commandArgs = await this.getCommandArgs();
+		if (commandArgs === undefined) {
+			return;
+		}
+
+		const commandActionPromise = this.runSuiteCloudCommand(commandArgs);
+		const commandMessage = this.translationService.getMessage(COMMAND.TRIGGERED, this.vscodeCommandName);
+		const statusBarMessage: string = this.translationService.getMessage(CREATE_FILE.MESSAGES.CREATING_FILE);
+		this.messageService.showInformationMessage(commandMessage, statusBarMessage, commandActionPromise);
+
+		const actionResult = await commandActionPromise;
+		if (actionResult.isSuccess()) {
+			const createdFileUri: Uri = Uri.file(actionResult.data.path);
+			workspace.openTextDocument(createdFileUri).then(
+				(createdFile: TextDocument) => {
+					window.showTextDocument(createdFile);
+				},
+				(error: any) => {
+					this.messageService.showCommandError(error);
+				}
+			);
+		} else {
+			this.messageService.showCommandError();
+		}
+	}
+
+	private async getCommandArgs(): Promise<{ [key: string]: string | string[] } | undefined> {
+		const args: { [key: string]: string | string[] } = {
+			project: this.getProjectFolderPath(),
+		};
+
+		const selectedScriptType = await this.promptScriptTypeQuestion();
+		if (selectedScriptType === undefined) {
+			return;
+		} else {
+			args.type = selectedScriptType.id;
+		}
+
+		const selectedModules = await this.promptAddModulesQuestion();
+		if (selectedModules === undefined) {
+			return;
+		} else if (Array.isArray(selectedModules) && selectedModules.length > 0) {
+			args.module = selectedModules;
+		}
+
+		const selectedFolder = await this.promptFolderSelection();
+		if (selectedFolder === undefined) {
+			return;
+		}
+		const fileName = await this.promptFileNameInputBox(selectedFolder);
+		if (fileName === undefined) {
+			return;
+		}
+
+		args.path = path.join(selectedFolder, fileName);
+
+		return args;
+	}
+
+	private promptScriptTypeQuestion(): Thenable<SuiteScriptTypeItem | undefined> {
+		return window.showQuickPick(
+			SUITESCRIPT_TYPES.map((el) => <SuiteScriptTypeItem>{ label: el.name, id: el.id, name: el.name }),
```

Comment:
could we give e1 a more informative name?

Reply:
I think SUITESCRIPT_TYPES is descriptive enough, it could also be SUITESCRIPT_TYPES_METADATA but I'm not sure.
What would be your suggestion?

Reply:
I'm refering to the el variable inside